crates/shared/src/price_estimation/sanitized.rs

```diff
@@ -389,139 +408,17 @@ mod tests {
-        assert_eq!(result.len(), 10);
-        assert_eq!(
-            result[0].as_ref().unwrap(),
-            &Estimate {
-                out_amount: 1.into(),
-                gas: 100,
-                solver: Default::default(),
-            }
-        );
-        assert_eq!(
-            result[1].as_ref().unwrap(),
-            &Estimate {
-                out_amount: 1.into(),
-                //sanitized_estimator will add ETH_UNWRAP_COST to the gas of any
-                //Query with ETH as the buy_token.
-                gas: GAS_PER_WETH_UNWRAP + 100,
-                solver: Default::default(),
-            }
-        );
-        assert!(matches!(
-            result[2].as_ref().unwrap_err(),
-            PriceEstimationError::ProtocolInternal(err)
-                if err.to_string() == "cost of converting native asset would overflow gas price",
-        ));
-        assert_eq!(
-            result[3].as_ref().unwrap(),
-            &Estimate {
-                out_amount: 1.into(),
-                //sanitized_estimator will add ETH_WRAP_COST to the gas of any
-                //Query with ETH as the sell_token.
-                gas: GAS_PER_WETH_WRAP + 100,
-                solver: Default::default(),
-            }
-        );
-        assert_eq!(
-            result[4].as_ref().unwrap(),
-            &Estimate {
-                out_amount: 1.into(),
-                gas: 0,
-                solver: Default::default(),
-            }
-        );
-        assert_eq!(
-            result[5].as_ref().unwrap(),
-            &Estimate {
-                out_amount: 1.into(),
-                gas: 0,
-                solver: Default::default(),
-            }
-        );
-        assert_eq!(
-            result[6].as_ref().unwrap(),
-            &Estimate {
-                out_amount: 1.into(),
-                // Sanitized estimator will report a 1:1 estimate when unwrapping native token.
-                gas: GAS_PER_WETH_UNWRAP,
-                solver: Default::default(),
-            }
-        );
-        assert_eq!(
-            result[7].as_ref().unwrap(),
-            &Estimate {
-                out_amount: 1.into(),
-                // Sanitized estimator will report a 1:1 estimate when wrapping native token.
-                gas: GAS_PER_WETH_WRAP,
-                solver: Default::default(),
+        for (query, expectation) in queries {
+            let result = sanitized_estimator.estimate(&query).await;
+            match result {
+                Ok(estimate) => assert_eq!(estimate, expectation.unwrap()),
+                Err(err) => {
+                    // we only compare the error variant; everything else would be a PITA
+                    let reported_error = std::mem::discriminant(&err);
+                    let expected_error = std::mem::discriminant(&expectation.unwrap_err());
+                    assert_eq!(reported_error, expected_error);
```

Comment:
This is a bit icky. We can't simply do `assert_eq!(result, expected)` because `anyhow::Error` does not implement `Eq`. As a work around we verify that the error variants are at least equal.